src/main/java/org/testng/internal/TestResult.java

```diff
@@ -230,7 +231,8 @@ private static String toString(int status) {
       case SKIP: return "SKIP";
       case SUCCESS_PERCENTAGE_FAILURE: return "SUCCESS WITHIN PERCENTAGE";
       case STARTED: return "STARTED";
-      default: throw new RuntimeException();
+      case CREATED: return "CREATED";
+      default: throw new TestNGException("Encountered an un-defined test status of [" + status + "].");

```

Comment:
I was thinking about converting these to `enum` as well but they are defined on `ITestResult`, which is part of the API, so I'm not 100% sure it's a backward compatible change.